### PR TITLE
feat: add support for disabling tracking via localStorage

### PR DESCRIPTION
**Pull request by supermar1010**
Added support for disabling tracking via localStorage(umami.disabled=1)  (#103)

### CODE REVIEW

**Comment by ijkml** [comment]
Thank you, Mario.
Looks great, I'll review and merge as soon as I can 👍.

**Comment by ijkml** [comment]
LGTM :+1:.
Changes I made were mostly to the playground (changed the button to a toggle).
Let's merge when you're ready.

File: internal/utils.ts [hunk 1]
```diff
@@ -102,15 +102,14 @@ const preflight = computed((): PreflightResult => {
   if (!host || !isValidHost(endpoint.value))
     return 'host';
 
-  const {
-    location: { hostname },
-  } = window;
+  const hostname = window.location.hostname;
+  const storage = window.localStorage;
```
Comment (supermar1010) [inline internal/utils.ts:106]: I'm honestly not sure what the difference in access between `window.localStorage` and just using `localStorage` is, but I also can't find any information on the internet about this, so if this also works, I don't have an oponion :D

**Comment by supermar1010** [comment]
Looks good to me (just see the comment I'm not sure about the difference between the localStorage call via window and without, but if it works it's fine for me!)

**Pull request merged**